src/cclib/parser/utils.py

```diff
@@ -194,6 +195,67 @@ def split(self, line, truncate=True):
         return elements
 
 
+class MoldenReformatter:
+    """Reformat Molden output files."""
+
+    def __init__(self, filestring):
+        self.filestring = filestring
+
+    def scinotation(self, num):
+        """Convert Molden style number formatting to scientific notation.
+        0.9910616900D+02 --> 9.910617e+01
+        """
+        num = num.replace('D', 'e')
+        return str('%.6e' % Decimal(num))
+
+    def reformat(self):
+        """Reformat Molden output file to:
+        - use scientific notation,
+        - split sp molecular orbitals to s and p, and
+        - replace multiple spaces with single."""
+        filelines = self.filestring.split("\n")
+        lines = []
+        # Flag to deterimine if the title section has been seen.
+        title = False
+        
+        for line in filelines:
+            line = line.replace('\n', '')
+            # Replace multiple spaces with single spaces.
+            line = ' '.join(line.split())
+
+            # Check for [Title] section.
+            if 'title' in line.lower():
+                lines.append(line)
+                # Add the next line to lines.
+                line = next(filelines).split()
+                line = ' '.join(line.split())
+                lines.append(line)
+                title = True
+
+            # Convert D notation to scientific notation.
+            if title and 'D' in line:
```

Comment:
Ok, now I see. It seems like the reformatting won't happen if the Title section is missing. Is that the case?

Also, does the D notation occur anywhere besides the GTO section? If not, then this block seems unnecessary since it'll be handled below.